src/floyd-warshall.jl

```diff
@@ -36,17 +36,17 @@ end
 #     Note that it is possible to consume large amounts of memory as the
 #     space required for the FloydWarshallState is O(n^2).
 #     """ ->
-function floyd_warshall_shortest_paths(
+function floyd_warshall_shortest_paths{T<:Real}(
     g::AbstractGraph;
-    edge_dists::AbstractArray{Float64, 2} = Array(Float64,(0,0))
+    edge_dists::AbstractArray{T,2} = zeros(0,0)
 )
 
     use_dists = issparse(edge_dists)? nnz(edge_dists > 0) : !isempty(edge_dists)
     n_v = nv(g)
-    dists = fill(convert(Float64,Inf), (n_v,n_v))
-    parents = zeros(Int, (n_v,n_v))
+    S = typeof(one(T) + one(T))
+    dists = fill(typemax(S), n_v, n_v)
```

Comment:
> Well, it turns out that this statement is false. 

Yes, it is - my apologies. This was a recent commit to LightGraphs for @jpfairbanks' work with spectral graph theory. The reason they're `Number` rather than  `Integer` (and this makes the case unequivocally for the proposed change we're discussing here) is because we wanted to be able to pass distance matrices - as well as adjacency matrices - in to create the graph.

Just out of interest, why can't one have complex distances?